### ISSUE TITLE
APIKEY.txt is not being generated

### ISSUE THREAD

**Issue by Essoufi-oussama** [ether/etherpad-lite#6350]
after installing on windows and running start.bat there is no apikey.txt being generated on the main folder

**Comment by jibon57** [comment]
Yes, from `v2.0.2` looks like oauth will be using by default. You can follow: https://github.com/ether/etherpad-lite/pull/6328

**Comment by SamTV12345** [comment]
> Yes, from now looks like `oauth` will be using by default. You can follow: #6328. But in `v2.0.2` none is working. Hope there will be a new release soon because fix was in this PR: #6325

Yes you are right. @webzwo0i and I are working on the docs under #6328 .

**Comment by SamTV12345** [comment]
> Yes, from now looks like `oauth` will be using by default. You can follow: #6328. But in `v2.0.2` none is working. Hope there will be a new release soon because fix was in this PR: #6325

I made it optionally. You can also continue to use the old APIKEY.txt file

**Comment by 12thDimension** [comment]
I put   "authenticationMethod": "${AUTHENTICATION_METHOD:apikey}" in my settings.json.docker, ran another "docker compose up -d" but the file is not being created for me. 
I checked the config in the admin panel but the setting wasnt even there

**Comment by SamTV12345** [comment]
> I put   "authenticationMethod": "${AUTHENTICATION_METHOD:apikey}" in my settings.json.docker, ran another "docker compose up -d" but the file is not being created for me. 
> 
> I checked the config in the admin panel but the setting wasnt even there

That means you are probably not mapping the settings.json file into the container. Is the path in your docker-compose really settings.docker.json instead of settings.json?

**Comment by 12thDimension** [comment]
I couldnt find any docu on this. I dont have a path in my docker-compose to my settings file. There's also no settings.json only settings.json.docker and settings.json.template

**Comment by timiil** [comment]
I have the same problem when using the docker-compose file. please tell us how to use the docker container that use APIKey.txt ??

**Comment by SamTV12345** [comment]
You can map the settings.json file. So when you have a file settings.json on your machine you can add a volume with ' ./settings.json:/opt/etherpad-lite/settings.json` . That way you have your settings.json linked in the container

**Comment by Dherlou** [comment]
TLDR for future readers, because it is not documented in the [docker.adoc](https://github.com/ether/etherpad-lite/blob/develop/doc/docker.adoc) yet:

It is sufficient to add the following to your `docker-compose.yml` to reenable legacy apikey authentication with a static API key. If you want to use a dynamically generated API key, omit the `volumes` part.
```yml
services:
  app:
    environment:
      AUTHENTICATION_METHOD: "apikey"
    volumes:
      - "<host-path-to>/APIKEY.txt:/opt/etherpad-lite/APIKEY.txt:ro"
```

**Comment by zentavr** [comment]
@Dherlou what should be the content of that file?

**Comment by Dherlou** [comment]
@zentavr

In the past, a secret key was generated on the first startup of Etherpad and put into the `APIKEY.txt`, if the authentication method was "apikey" and the file did not exist yet. (see https://etherpad.org/doc/v1.9.1/#_authentication)

This does not seem to be the case anymore, probably because this API key authentication is deprecated now and they are slowly phasing it out, so just create this text file with an arbitrary string (like a password) as its content and mount it inside the Etherpad container as shown above or better with a docker volume instead of a bind mount. This content / key is what you use for API requests then.

**Comment by SamTV12345** [comment]
Adding to this we haven't decided on a deprecation plan. It should be generated.

**Comment by zentavr** [comment]
@Dherlou I’d created a helm chart for deploying the service into Kubernetes and use read only filesystem when deploying the image. I’ve got a lot of errors when the container started, thats why I’m asking.

I was able to run it with a couple of workarounds.
